x-pack/server/lib/create_router/index.js

```diff
@@ -16,13 +16,13 @@ export const createRouter = (server, pluginId, apiBasePath = '') => {
   const requestHandler = (handler) => async (request, h) => {
     const callWithRequest = callWithRequestFactory(server, request);
     try {
-      return handler(request, callWithRequest, h);
+      return await handler(request, callWithRequest, h);
     } catch (err) {
       if (isEsError(err)) {
-        throw wrapEsError(err);
+        return wrapEsError(err);
```

Comment:
usage of `throw` wasn't an issue, but I changed it to `return` since that is more commonly used to return errors in x-pack server code

Reply:
@jen-huang Just seeing this. Have you tested this? It seems that by not throwing we would never be able to catch an error...